src/dawn/Optimizer/DependencyGraph.h

```diff
@@ -33,8 +35,15 @@ namespace dawn {
 
 /// @brief CRTP base class of all dependency graphs
 /// @ingroup optimizer
-template <class Derived, class EdgeData>
+template <class Derived, class EdgeData, class VertexData>
 class DependencyGraph {
+
+  template <typename IntT, typename VertexDataT>
```

Comment:
yes, this IntT is coming from the templated method using `vertex_has_data`

https://github.com/cosunae/dawn/pull/3/files#diff-d2881d3937c88dd6eb9495bccde142f7R107

We are using SFINAE (enable_if) in order to customize the signature of the API according to whether the Vertex type has additional data. Enable_if will only work if we check on a templated type. If the function signature can not be resolved instantiating the template as an specific type, the compiler will then not instantiate that signature, but wont be an error, and will continue looking for a signature that matches. However the type must be templated, and not a fully instantiated type. Otherwise it is an error. Sinve VertexData is not a template we need to template the only argument we have in the function, ID, which is int. So IntT is just a templatization of int